staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -5592,10 +5592,12 @@ const (
 	// TODO: Consider supporting different formats, specifying CA/destinationCA.
 	SecretTypeTLS SecretType = "kubernetes.io/tls"
 
-	// TLSCertKey is the key for tls certificates in a TLS secert.
+	// TLSCertKey is the key for TLS certificates in a TLS secret.
 	TLSCertKey = "tls.crt"
 	// TLSPrivateKeyKey is the key for the private key field in a TLS secret.
 	TLSPrivateKeyKey = "tls.key"
+	// TLSCABundleKey is the certificate bundle from the CA that issued the TLS certificate.
```

Comment:
this isn't typically what a ca.crt means... I have similar questions to what I raised in https://github.com/kubernetes/kubernetes/pull/91222#issuecomment-630531596

When an actor is consuming a cert/key pair in order to make a TLS connection, the cert/key is typically what they present to the other party, and the ca.crt bundle is what they use to verify the certificate presented by the other party. There is no inherent relationship between the CAs that issued the two certificates (and in fact it is common for different CAs to issue client and serving certs). I'm not sure bundling the CA used to verify the remote connection with the cert/key presented to assert identity makes sense.

Reply:
I was approaching this from the server certificate perspective, so the CA bundle would be the intermediate certificates leading from the certificate in the secret to the trusted root. If you are doing TLS client auth, I think that this makes sense too, since the server may need the client's intermediate certificates to validate its chain.

This field isn't intended to be the CA that you validate a TLS peer against. Maybe the description needs some additional language to make that clearer.

Reply:
> This field isn't intended to be the CA that you validate a TLS peer against

It's already being used that way by some clients. See https://kubernetes.github.io/ingress-nginx/examples/auth/client-certs/

Reply:
I'm not that familiar with the code but it looks like ingress-nginx supports the key [generically](https://github.com/kubernetes/ingress-nginx/blob/master/internal/ingress/controller/store/backend_ssl.go#L37).

Maybe I should have said "my intention in proposing this key is not that it be the CA that you validate the TLS peer against" :)

If the key is already used for conflicting purposes, maybe updating the language here to be more neutral and just say that it is used for a CA certificates?